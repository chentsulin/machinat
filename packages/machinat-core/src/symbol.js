// @flow
export const MACHINAT_ELEMENT_TYPE = Symbol('machinat.element');

export const MACHINAT_FRAGMENT_TYPE = Symbol('machinat.element.fragment');

export const MACHINAT_PAUSE_TYPE = Symbol('machinat.element.pause');

export const MACHINAT_PROVIDER_TYPE = Symbol('machinat.element.provider');

export const MACHINAT_THUNK_TYPE = Symbol('machinat.element.thunk');

export const MACHINAT_RAW_TYPE = Symbol('machinat.element.raw');

export const MACHINAT_NATIVE_TYPE = Symbol('machinat.component.native');

export const MACHINAT_SERVICES_CONTAINER = Symbol(
  'machinat.services.container'
);

export const MACHINAT_SERVICES_INTERFACE = Symbol(
  'machinat.services.interface'
);

export const MACHINAT_SERVICES_PROVIDER = Symbol('machinat.services.provider');
