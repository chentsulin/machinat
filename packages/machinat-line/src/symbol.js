// @flow
export const LINE_NAITVE_TYPE = Symbol.for('machinat.line.native');

export const NO_RENDERED = Symbol.for('machinat.line.no_rendered');
