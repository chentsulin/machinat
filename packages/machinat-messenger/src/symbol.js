export const MESSENGER_NAITVE_TYPE = Symbol.for('machinat.messenger.native');
