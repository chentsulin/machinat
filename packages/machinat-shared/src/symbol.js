export const MACHINAT_ELEMENT_TYPE = Symbol.for('machinat.element');

export const MACHINAT_ROOT_ELEMENT = Symbol.for('machinat.element.root');

export const MACHINAT_FRAGMENT_TYPE = Symbol.for('machinat.element.fragment');

export const MACHINAT_ASYNC_TYPE = Symbol.for('machinat.element.async');
