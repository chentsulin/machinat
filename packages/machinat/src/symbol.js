// @flow
export const MACHINAT_ELEMENT_TYPE = Symbol.for('machinat.element');

export const MACHINAT_FRAGMENT_TYPE = Symbol.for('machinat.element.fragment');

export const MACHINAT_PAUSE_TYPE = Symbol.for('machinat.element.pause');

export const MACHINAT_NATIVE_TYPE = Symbol.for('machinat.component.native');

export const MACHINAT_SERVICE_TYPE = Symbol.for('machinat.service');

export const MACHINAT_PROVIDER_TYPE = Symbol.for('machinat.service.provider');

export const MACHINAT_CONSUMER_TYPE = Symbol.for('machinat.service.consumer');

export const SEGMENT_BREAK = Symbol.for('machinat.segment.break');
