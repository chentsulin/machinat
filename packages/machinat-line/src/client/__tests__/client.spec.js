import moxy from 'moxy';
import nock from 'nock';
import Machinat from 'machinat';
import LineClient from '../client';
import { LINE_NAITVE_TYPE } from '../../symbol';

nock.disableNetConnect();

const accessToken = '__LINE_CHANNEL_TOKEN__';

const Foo = moxy(props => [props]);
Foo.$$native = LINE_NAITVE_TYPE;
Foo.$$unit = true;

const Bar = moxy(props => [props]);
Bar.$$native = LINE_NAITVE_TYPE;
Bar.$$unit = true;
Bar.$$entry = (thread, value) => `${thread.source.userId}/bar/${value.id}`;
Bar.$$hasBody = false;

const Baz = moxy(props => [props]);
Baz.$$native = LINE_NAITVE_TYPE;
Baz.$$unit = true;
Baz.$$entry = (thread, value) => `${thread.source.userId}/baz/${value.id}`;
Baz.$$hasBody = true;

const msg = (
  <>
    <Foo id={1} />
    <Foo id={2} />
    <Foo id={3} />
    <Foo id={4} />
    <Foo id={5} />
    <Foo id={6} />
    <Machinat.Pause />
    <Foo id={7} />
    <Bar id={8} />
    <Foo id={9} />
    <Baz id={10} />
  </>
);

let lineAPI;
beforeEach(() => {
  lineAPI = nock('https://api.line.me', {
    reqheaders: {
      'content-type': 'application/json',
      authorization: 'Bearer __LINE_CHANNEL_TOKEN__',
    },
  });
});

it('pushs message ok', async () => {
  const client = new LineClient({
    accessToken,
    useReplyAPI: false,
    consumeInterval: 500,
  });

  const thread = { type: 'user', userId: 'foo' };
  const promise = client.send(thread, msg);

  const msgScope = lineAPI
    .post('/v2/bot/message/push', body => {
      expect(body.to).toBe('foo');
      expect(body.replyToken).toBe(undefined);

      expect(body).toMatchSnapshot();
      return true;
    })
    .times(4)
    .reply(200, '{}');

  const barScope = lineAPI
    .post('/v2/bot/foo/bar/8', body => {
      expect(body).toBe('');
      return true;
    })
    .reply(200, '{}');

  const bazScope = lineAPI
    .post('/v2/bot/foo/baz/10', body => {
      expect(body).toEqual({ id: 10 });
      return true;
    })
    .reply(200, '{}');

  client.startConsumingJob();
  const response = await promise;

  expect(msgScope.isDone()).toBe(true);
  expect(barScope.isDone()).toBe(true);
  expect(bazScope.isDone()).toBe(true);

  expect(response).toMatchSnapshot();
});

it('reply message ok', async () => {
  const client = new LineClient({
    accessToken,
    useReplyAPI: true,
    consumeInterval: 500,
  });

  const thread = { type: 'user', userId: 'foo' };
  const promise = client.send(thread, msg, { replyToken: '__REPLY_TOKEN__' });

  const msgScope = lineAPI
    .post('/v2/bot/message/reply', body => {
      expect(body.to).toBe(undefined);
      expect(body.replyToken).toBe('__REPLY_TOKEN__');

      expect(body).toMatchSnapshot();
      return true;
    })
    .times(4)
    .reply(200, '{}');

  const barScope = lineAPI
    .post('/v2/bot/foo/bar/8', body => {
      expect(body).toBe('');
      return true;
    })
    .reply(200, '{}');

  const bazScope = lineAPI
    .post('/v2/bot/foo/baz/10', body => {
      expect(body).toEqual({ id: 10 });
      return true;
    })
    .reply(200, '{}');

  client.startConsumingJob();
  const response = await promise;

  expect(msgScope.isDone()).toBe(true);
  expect(barScope.isDone()).toBe(true);
  expect(bazScope.isDone()).toBe(true);

  expect(response).toMatchSnapshot();
});

it('throw if replyToken not given when useReplyAPI set to true', async () => {
  const client = new LineClient({
    accessToken,
    useReplyAPI: true,
    consumeInterval: 500,
  });

  const thread = { type: 'user', userId: 'foo' };
  expect(client.send(thread, msg)).rejects.toThrowErrorMatchingInlineSnapshot(
    `"replyToken option must be given while sending when useReplyAPI set to true"`
  );
});

it('throw if connection error happen', async () => {
  const client = new LineClient({
    accessToken,
    useReplyAPI: false,
    consumeInterval: 500,
  });

  const scope = lineAPI
    .post('/v2/bot/message/push')
    .replyWithError('something wrong like connection error');

  client.startConsumingJob();
  await expect(client.send({ type: 'user', userId: 'foo' }, msg)).rejects
    .toThrowErrorMatchingInlineSnapshot(`
"Errors happen while sending:

	FetchError: request to https://api.line.me/v2/bot/message/push failed, reason: something wrong like connection error
    at OverriddenClientRequest.<anonymous> (/Users/LR/Documents/machinat/node_modules/node-fetch/lib/index.js:1345:11)
    at OverriddenClientRequest.emit (events.js:182:13)
    at /Users/LR/Documents/machinat/node_modules/nock/lib/request_overrider.js:221:11
    at process._tickCallback (internal/process/next_tick.js:61:11)"
`);

  expect(scope.isDone()).toBe(true);
});

it('throw if api error happen', async () => {
  const client = new LineClient({
    accessToken,
    useReplyAPI: false,
    consumeInterval: 500,
  });

  const scope1 = lineAPI.post('/v2/bot/message/push').reply(200, {});
  const scope2 = lineAPI.post('/v2/bot/message/push').reply(400, {
    message: 'The request body has 2 error(s)',
    details: [
      {
        message: 'May not be empty',
        property: 'messages[0].text',
      },
      {
        message:
          'Must be one of the following values: [text, image, video, audio, location, sticker, template, imagemap]',
        property: 'messages[1].type',
      },
    ],
  });

  client.startConsumingJob();
  await expect(client.send({ type: 'user', userId: 'foo' }, msg)).rejects
    .toThrowErrorMatchingInlineSnapshot(`
"Errors happen while sending:

	Bad Request: The request body has 2 error(s). 1) May not be empty, at messages[0].text. 2) Must be one of the following values: [text, image, video, audio, location, sticker, template, imagemap], at messages[1].type.
    at LineClient._request (/Users/LR/Documents/machinat/packages/machinat-line/src/client/client.js:138:13)
    at process._tickCallback (internal/process/next_tick.js:68:7)"
`);

  expect(scope1.isDone()).toBe(true);
  expect(scope2.isDone()).toBe(true);
});
