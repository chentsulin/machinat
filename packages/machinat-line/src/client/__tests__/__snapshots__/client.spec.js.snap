// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pushs message ok 1`] = `
Object {
  "messages": Array [
    Object {
      "id": 1,
    },
    Object {
      "id": 2,
    },
    Object {
      "id": 3,
    },
    Object {
      "id": 4,
    },
    Object {
      "id": 5,
    },
  ],
  "to": "foo",
}
`;

exports[`pushs message ok 2`] = `
Object {
  "messages": Array [
    Object {
      "id": 6,
    },
  ],
  "to": "foo",
}
`;

exports[`pushs message ok 3`] = `
Object {
  "messages": Array [
    Object {
      "id": 7,
    },
  ],
  "to": "foo",
}
`;

exports[`pushs message ok 4`] = `
Object {
  "messages": Array [
    Object {
      "id": 9,
    },
  ],
  "to": "foo",
}
`;

exports[`pushs message ok 5`] = `
Object {
  "actions": Array [
    Object {
      "asUnit": true,
      "element": Object {
        "$$typeof": Symbol(machinat.element),
        "props": Object {
          "id": 1,
        },
        "type": [Function],
      },
      "isPause": false,
      "path": "$::0",
      "value": Object {
        "id": 1,
      },
    },
    Object {
      "asUnit": true,
      "element": Object {
        "$$typeof": Symbol(machinat.element),
        "props": Object {
          "id": 2,
        },
        "type": [Function],
      },
      "isPause": false,
      "path": "$::1",
      "value": Object {
        "id": 2,
      },
    },
    Object {
      "asUnit": true,
      "element": Object {
        "$$typeof": Symbol(machinat.element),
        "props": Object {
          "id": 3,
        },
        "type": [Function],
      },
      "isPause": false,
      "path": "$::2",
      "value": Object {
        "id": 3,
      },
    },
    Object {
      "asUnit": true,
      "element": Object {
        "$$typeof": Symbol(machinat.element),
        "props": Object {
          "id": 4,
        },
        "type": [Function],
      },
      "isPause": false,
      "path": "$::3",
      "value": Object {
        "id": 4,
      },
    },
    Object {
      "asUnit": true,
      "element": Object {
        "$$typeof": Symbol(machinat.element),
        "props": Object {
          "id": 5,
        },
        "type": [Function],
      },
      "isPause": false,
      "path": "$::4",
      "value": Object {
        "id": 5,
      },
    },
    Object {
      "asUnit": true,
      "element": Object {
        "$$typeof": Symbol(machinat.element),
        "props": Object {
          "id": 6,
        },
        "type": [Function],
      },
      "isPause": false,
      "path": "$::5",
      "value": Object {
        "id": 6,
      },
    },
    Object {
      "asUnit": true,
      "element": Object {
        "$$typeof": Symbol(machinat.element),
        "props": Object {},
        "type": Symbol(machinat.element.pause),
      },
      "isPause": true,
      "path": "$::6",
      "value": undefined,
    },
    Object {
      "asUnit": true,
      "element": Object {
        "$$typeof": Symbol(machinat.element),
        "props": Object {
          "id": 7,
        },
        "type": [Function],
      },
      "isPause": false,
      "path": "$::7",
      "value": Object {
        "id": 7,
      },
    },
    Object {
      "asUnit": true,
      "element": Object {
        "$$typeof": Symbol(machinat.element),
        "props": Object {
          "id": 8,
        },
        "type": [Function],
      },
      "isPause": false,
      "path": "$::8",
      "value": Object {
        "id": 8,
      },
    },
    Object {
      "asUnit": true,
      "element": Object {
        "$$typeof": Symbol(machinat.element),
        "props": Object {
          "id": 9,
        },
        "type": [Function],
      },
      "isPause": false,
      "path": "$::9",
      "value": Object {
        "id": 9,
      },
    },
    Object {
      "asUnit": true,
      "element": Object {
        "$$typeof": Symbol(machinat.element),
        "props": Object {
          "id": 10,
        },
        "type": [Function],
      },
      "isPause": false,
      "path": "$::10",
      "value": Object {
        "id": 10,
      },
    },
  ],
  "jobs": Array [
    Object {
      "apiEntry": "message/push",
      "body": Object {
        "messages": Array [
          Object {
            "id": 1,
          },
          Object {
            "id": 2,
          },
          Object {
            "id": 3,
          },
          Object {
            "id": 4,
          },
          Object {
            "id": 5,
          },
        ],
        "to": "foo",
      },
      "hasBody": true,
      "threadId": "line:user:foo",
    },
    Object {
      "apiEntry": "message/push",
      "body": Object {
        "messages": Array [
          Object {
            "id": 6,
          },
        ],
        "to": "foo",
      },
      "hasBody": true,
      "threadId": "line:user:foo",
    },
    Object {
      "apiEntry": "message/push",
      "body": Object {
        "messages": Array [
          Object {
            "id": 7,
          },
        ],
        "to": "foo",
      },
      "hasBody": true,
      "threadId": "line:user:foo",
    },
    Object {
      "apiEntry": "foo/bar/8",
      "body": Object {
        "id": 8,
      },
      "hasBody": false,
      "threadId": "line:user:foo",
    },
    Object {
      "apiEntry": "message/push",
      "body": Object {
        "messages": Array [
          Object {
            "id": 9,
          },
        ],
        "to": "foo",
      },
      "hasBody": true,
      "threadId": "line:user:foo",
    },
    Object {
      "apiEntry": "foo/baz/10",
      "body": Object {
        "id": 10,
      },
      "hasBody": true,
      "threadId": "line:user:foo",
    },
  ],
  "message": Object {
    "$$typeof": Symbol(machinat.element),
    "props": Object {
      "children": Array [
        Object {
          "$$typeof": Symbol(machinat.element),
          "props": Object {
            "id": 1,
          },
          "type": [Function],
        },
        Object {
          "$$typeof": Symbol(machinat.element),
          "props": Object {
            "id": 2,
          },
          "type": [Function],
        },
        Object {
          "$$typeof": Symbol(machinat.element),
          "props": Object {
            "id": 3,
          },
          "type": [Function],
        },
        Object {
          "$$typeof": Symbol(machinat.element),
          "props": Object {
            "id": 4,
          },
          "type": [Function],
        },
        Object {
          "$$typeof": Symbol(machinat.element),
          "props": Object {
            "id": 5,
          },
          "type": [Function],
        },
        Object {
          "$$typeof": Symbol(machinat.element),
          "props": Object {
            "id": 6,
          },
          "type": [Function],
        },
        Object {
          "$$typeof": Symbol(machinat.element),
          "props": Object {},
          "type": Symbol(machinat.element.pause),
        },
        Object {
          "$$typeof": Symbol(machinat.element),
          "props": Object {
            "id": 7,
          },
          "type": [Function],
        },
        Object {
          "$$typeof": Symbol(machinat.element),
          "props": Object {
            "id": 8,
          },
          "type": [Function],
        },
        Object {
          "$$typeof": Symbol(machinat.element),
          "props": Object {
            "id": 9,
          },
          "type": [Function],
        },
        Object {
          "$$typeof": Symbol(machinat.element),
          "props": Object {
            "id": 10,
          },
          "type": [Function],
        },
      ],
    },
    "type": Symbol(machinat.element.fragment),
  },
  "results": Array [
    Object {},
    Object {},
    Object {},
    Object {},
    Object {},
    Object {},
  ],
}
`;

exports[`reply message ok 1`] = `
Object {
  "messages": Array [
    Object {
      "id": 1,
    },
    Object {
      "id": 2,
    },
    Object {
      "id": 3,
    },
    Object {
      "id": 4,
    },
    Object {
      "id": 5,
    },
  ],
  "replyToken": "__REPLY_TOKEN__",
}
`;

exports[`reply message ok 2`] = `
Object {
  "messages": Array [
    Object {
      "id": 6,
    },
  ],
  "replyToken": "__REPLY_TOKEN__",
}
`;

exports[`reply message ok 3`] = `
Object {
  "messages": Array [
    Object {
      "id": 7,
    },
  ],
  "replyToken": "__REPLY_TOKEN__",
}
`;

exports[`reply message ok 4`] = `
Object {
  "messages": Array [
    Object {
      "id": 9,
    },
  ],
  "replyToken": "__REPLY_TOKEN__",
}
`;

exports[`reply message ok 5`] = `
Object {
  "actions": Array [
    Object {
      "asUnit": true,
      "element": Object {
        "$$typeof": Symbol(machinat.element),
        "props": Object {
          "id": 1,
        },
        "type": [Function],
      },
      "isPause": false,
      "path": "$::0",
      "value": Object {
        "id": 1,
      },
    },
    Object {
      "asUnit": true,
      "element": Object {
        "$$typeof": Symbol(machinat.element),
        "props": Object {
          "id": 2,
        },
        "type": [Function],
      },
      "isPause": false,
      "path": "$::1",
      "value": Object {
        "id": 2,
      },
    },
    Object {
      "asUnit": true,
      "element": Object {
        "$$typeof": Symbol(machinat.element),
        "props": Object {
          "id": 3,
        },
        "type": [Function],
      },
      "isPause": false,
      "path": "$::2",
      "value": Object {
        "id": 3,
      },
    },
    Object {
      "asUnit": true,
      "element": Object {
        "$$typeof": Symbol(machinat.element),
        "props": Object {
          "id": 4,
        },
        "type": [Function],
      },
      "isPause": false,
      "path": "$::3",
      "value": Object {
        "id": 4,
      },
    },
    Object {
      "asUnit": true,
      "element": Object {
        "$$typeof": Symbol(machinat.element),
        "props": Object {
          "id": 5,
        },
        "type": [Function],
      },
      "isPause": false,
      "path": "$::4",
      "value": Object {
        "id": 5,
      },
    },
    Object {
      "asUnit": true,
      "element": Object {
        "$$typeof": Symbol(machinat.element),
        "props": Object {
          "id": 6,
        },
        "type": [Function],
      },
      "isPause": false,
      "path": "$::5",
      "value": Object {
        "id": 6,
      },
    },
    Object {
      "asUnit": true,
      "element": Object {
        "$$typeof": Symbol(machinat.element),
        "props": Object {},
        "type": Symbol(machinat.element.pause),
      },
      "isPause": true,
      "path": "$::6",
      "value": undefined,
    },
    Object {
      "asUnit": true,
      "element": Object {
        "$$typeof": Symbol(machinat.element),
        "props": Object {
          "id": 7,
        },
        "type": [Function],
      },
      "isPause": false,
      "path": "$::7",
      "value": Object {
        "id": 7,
      },
    },
    Object {
      "asUnit": true,
      "element": Object {
        "$$typeof": Symbol(machinat.element),
        "props": Object {
          "id": 8,
        },
        "type": [Function],
      },
      "isPause": false,
      "path": "$::8",
      "value": Object {
        "id": 8,
      },
    },
    Object {
      "asUnit": true,
      "element": Object {
        "$$typeof": Symbol(machinat.element),
        "props": Object {
          "id": 9,
        },
        "type": [Function],
      },
      "isPause": false,
      "path": "$::9",
      "value": Object {
        "id": 9,
      },
    },
    Object {
      "asUnit": true,
      "element": Object {
        "$$typeof": Symbol(machinat.element),
        "props": Object {
          "id": 10,
        },
        "type": [Function],
      },
      "isPause": false,
      "path": "$::10",
      "value": Object {
        "id": 10,
      },
    },
  ],
  "jobs": Array [
    Object {
      "apiEntry": "message/reply",
      "body": Object {
        "messages": Array [
          Object {
            "id": 1,
          },
          Object {
            "id": 2,
          },
          Object {
            "id": 3,
          },
          Object {
            "id": 4,
          },
          Object {
            "id": 5,
          },
        ],
        "replyToken": "__REPLY_TOKEN__",
      },
      "hasBody": true,
      "threadId": "line:user:foo",
    },
    Object {
      "apiEntry": "message/reply",
      "body": Object {
        "messages": Array [
          Object {
            "id": 6,
          },
        ],
        "replyToken": "__REPLY_TOKEN__",
      },
      "hasBody": true,
      "threadId": "line:user:foo",
    },
    Object {
      "apiEntry": "message/reply",
      "body": Object {
        "messages": Array [
          Object {
            "id": 7,
          },
        ],
        "replyToken": "__REPLY_TOKEN__",
      },
      "hasBody": true,
      "threadId": "line:user:foo",
    },
    Object {
      "apiEntry": "foo/bar/8",
      "body": Object {
        "id": 8,
      },
      "hasBody": false,
      "threadId": "line:user:foo",
    },
    Object {
      "apiEntry": "message/reply",
      "body": Object {
        "messages": Array [
          Object {
            "id": 9,
          },
        ],
        "replyToken": "__REPLY_TOKEN__",
      },
      "hasBody": true,
      "threadId": "line:user:foo",
    },
    Object {
      "apiEntry": "foo/baz/10",
      "body": Object {
        "id": 10,
      },
      "hasBody": true,
      "threadId": "line:user:foo",
    },
  ],
  "message": Object {
    "$$typeof": Symbol(machinat.element),
    "props": Object {
      "children": Array [
        Object {
          "$$typeof": Symbol(machinat.element),
          "props": Object {
            "id": 1,
          },
          "type": [Function],
        },
        Object {
          "$$typeof": Symbol(machinat.element),
          "props": Object {
            "id": 2,
          },
          "type": [Function],
        },
        Object {
          "$$typeof": Symbol(machinat.element),
          "props": Object {
            "id": 3,
          },
          "type": [Function],
        },
        Object {
          "$$typeof": Symbol(machinat.element),
          "props": Object {
            "id": 4,
          },
          "type": [Function],
        },
        Object {
          "$$typeof": Symbol(machinat.element),
          "props": Object {
            "id": 5,
          },
          "type": [Function],
        },
        Object {
          "$$typeof": Symbol(machinat.element),
          "props": Object {
            "id": 6,
          },
          "type": [Function],
        },
        Object {
          "$$typeof": Symbol(machinat.element),
          "props": Object {},
          "type": Symbol(machinat.element.pause),
        },
        Object {
          "$$typeof": Symbol(machinat.element),
          "props": Object {
            "id": 7,
          },
          "type": [Function],
        },
        Object {
          "$$typeof": Symbol(machinat.element),
          "props": Object {
            "id": 8,
          },
          "type": [Function],
        },
        Object {
          "$$typeof": Symbol(machinat.element),
          "props": Object {
            "id": 9,
          },
          "type": [Function],
        },
        Object {
          "$$typeof": Symbol(machinat.element),
          "props": Object {
            "id": 10,
          },
          "type": [Function],
        },
      ],
    },
    "type": Symbol(machinat.element.fragment),
  },
  "results": Array [
    Object {},
    Object {},
    Object {},
    Object {},
    Object {},
    Object {},
  ],
}
`;
